nltk/tag/stanford.py

```diff
@@ -204,6 +205,79 @@ def parse_output(self, text, sentences):
 
         raise NotImplementedError
 
+class CoreNLPTagger(CoreNLPParser, TaggerI):
+    def __init__(self, url='http://localhost:9000', encoding='utf8'):
+        """
+        This is a abstract, duck-type of the CoreNLP that returns the
+        POS/NER tags from the Stanford CoreNLP API at nltk.parse.corenlp.
+        """
+        super(CoreNLPTagger, self).__init__(url, encoding)
+
+    def tag_sents(self, sentences, tagtype):
+        sentences = (' '.join(words) for words in sentences)
+        return list(self.raw_tag_sents(sentences, tagtype))
+
+
+    def tag(self, sentence, tagtype):
+        return self.tag_sents([sentence], tagtype)[0]
+
+    def raw_tag_sents(self, sentences, tagtype):
+        """
+        This function will interface the `GenericCoreNLPParser.api_call` to
+        retreive the JSON output and return the annotations required.
+        """
+        default_properties = {'ssplit.isOneSentence': 'true',
+                              'annotators': 'tokenize,' }
+        # Supports only 'pos' or 'ner' tags.
+        assert tagtype in ['pos', 'ner']
+        default_properties['annotators'] += tagtype
+        for sentence in sentences:
+            tagged_data = self.api_call(sentence, properties=default_properties)
+            assert len(tagged_data['sentences']) == 1
+            # Taggers only need to return 1-best sentence.
+            yield [(token['word'], token[tagtype]) for token in tagged_data['sentences'][0]['tokens']]
+
+
+class CoreNLPPOSTagger(CoreNLPTagger):
+    """
+    This is a duckling-type of the CoreNLPTagger that wraps around the
+    nltk.parse.CoreNLPParser for Part-of-Sppech tagging.
+
+        >>> from nltk.tag.stanford import CoreNLPPOSTagger
+        >>> tagged = CoreNLPPOSTagger().tag('What is the airspeed of an unladen swallow ?'.split())   # doctest: +SKIP
+        >>> expected = [('What', 'WP'), ('is', 'VBZ'), ('the', 'DT'),
+        ... ('airspeed', 'NN'), ('of', 'IN'), ('an', 'DT'), ('unladen', 'JJ'),
+        ... ('swallow', 'VB'), ('?', '.')]
+        >>> expected = tagged # doctest: +SKIP
+        True
+    """
+    def tag_sents(self, sentences, tagtype='pos'):
```

Comment:
I would define a tagtype as a class member and then the CoreNLPTagger.tag_sents would check it. Same goes for tag().